### DISCUSSION TITLE
Directus new Azure SQL server issues

### DISCUSSION THREAD

**Discussion by dialing-up** [directus/directus#4046] [Archive]
Hello everyone, I need some help interfacing Directus with Azure SQL Database. When I try to link it to a new database, I get
 
```
[EREQUEST] ALTER TABLE [directus_fields] ADD CONSTRAINT [directus_fields_group_foreign] FOREIGN KEY ([group]) REFERENCES [directus_fields] ([id]) ON UPDATE CASCADE ON DELETE SET NULL - Could not create constraint or index. See previous errors.

Please try again
```

and if I try again I get

```
Something went wrong while seeding the database:

[Error] Database is already installed

```
And if I try to start Directus now I get 

```
npx directus start
Error: knex: Required configuration option 'client' is missing.
```

Can anyone please help me here, I am just starting out

I think the issue is that Azure SQL wants `ON UPDATE NO ACTION ON DELETE NO ACTION` instead of `ON UPDATE CASCADE ON ON DELETE SET NULL`, if I run that instead in the Azure Query builder it creates those tables but does not add the columns required by directus, so afterwards running directus by installing it and changing the variables in the .env file just shows a white page.

If you could tell in which file in the create-directus-project is the SQL script so I can manually change it, I think that will solve the problem. But I don't know, this is my first time doing any of this.

**Comment by philleepflorence** [comment]
Hello @dialing-up - I am not yet fully versed in the API core code yet, but let's start with this:

What method did you use to install Directus?
Did you connect successfully to the Azure DB Instance, before getting the first error?

`Error: knex: Required configuration option 'client' is missing.`
Most Common Reason:
Missing or Unrecognizable `DB_CLIENT` variable in the API `.env` file - See [Environment Variables - Database](https://docs.directus.io/reference/environment-variables/#database)...
Also, see this related [discussion](https://github.com/directus/directus/discussions/3532).

There are also other discussions related to Azure - See [Discussions containing Azure](https://github.com/directus/directus/discussions?discussions_q=azure)

Directus depends on [Knex.js](http://knexjs.org/) to interface with the supported [SQL Database Vendors](https://docs.directus.io/guides/installation/cli/#_1-confirm-minimum-requirements).

**Reply by dialing-up** [reply]
1. I used create-directus-project
2. I connected successfully to Azure SQL DB, it doesn’t finish all the necessary tables and columns 
3. When I add the .env file and launch the server, it shows a blank page, and in the console and terminal it shows error messages that it could not fetch resources

**Reply by philleepflorence** [reply]
Following (while also validating) the Directus Docs:

> I am assuming Knex is composing the SQL build queries...
I did not discover any vendor specific queries in `/api/src/database`.

* Could any of these issues be related: [Knex Open Issues that contain "MSSQL ON DELETE"](https://github.com/knex/knex/issues?q=is%3Aissue+is%3Aopen+MSSQL+ON+DELETE)
* When did you add the .env file to the project, before or after `...connected successfully to Azure SQL DB...`, while using [Installing from CLI - Create a Project](https://docs.directus.io/guides/installation/cli/#_2-create-a-project).
* What is a good resource for validating `ON UPDATE CASCADE ON ON DELETE SET NULL` for MSSQL, if it is different from the other SQL systems.
* Are the terms **MS-SQL Server** in [Confirm Minimum Requirements](https://docs.directus.io/guides/installation/cli/#_1-confirm-minimum-requirements), **MSSQL** on [KnexJs](http://knexjs.org/), and **Azure SQL** on [What is Azure SQL](https://docs.microsoft.com/en-us/azure/azure-sql/azure-sql-iaas-vs-paas-what-is-overview) interchangeable and all variants of [SQL](https://www.w3schools.com/sql/)?.

**Comment by biapar** [comment]
Any news on this?
